MultiLangProgGuide/MultiLangProgGuide.rst

```diff
@@ -1819,6 +1821,920 @@ If the requested :code:`timeout_nsec` exceeds
+section on `Transaction Processing`_ for details.
+
+Since Go does not permit a pointer to a Go function to be passed as a
+parameter, :code:`tpfn` is a pointer to a C function that implements
+logic for the transaction. For a Go application, the C function is an
+interlude that in turn calls the Go function. YottaDB provides a shell
+script `GenYDBInterlude.sh`_ to generate interlude functions.
+
+A function implementing logic for a transaction should return one of
+the following:
+
+- :code:`C.YDB_OK` to indicate that per application logic, the
+  transaction can be committed. The YottaDB database engine will
+  commit the transaction if it is able to, as discussed in
+  `Transaction Processing`_, and if not, will call the function again.
+- :code:`C.YDB_TPRESTART` to indicate that the transaction should
+  restart, either because application logic has so determined or
+  because a YottaDB function called by the function has returned
+  :code:`C.YDB_TPRESTART`.
+- :code:`C.YDB_ROLLBACK` to indicate that :code:`TpS()` should not
+  commit the transaction.
+
+Also, in order to provide the function implementing the transaction
+logic with a parameter or parameters, :code:`tpfnparm` is passed to
+the interlude function and can in turn be passed to a Go function
+called by the interlude function.
+
+:code:`BufferTArray` is a list of local variables whose values
+should be saved, and restored to their original values when the
+transaction restarts. If the number of used array elements (the value
+returned by the :code:`GetUsed()` method) is zero, no local variables
+are saved and restored, and if the number of used array elements is 1,
+and that sole element references the string "*" all local variables
+are saved and restored.
+
+A case-insensitive value of "BA" or "BATCH" for :code:`transid`
+indicates to YottaDB that it need not ensure Durability for this
+transaction (it continues to ensure Atomicity, Consistency, and
+Isolation), as discussed under `ydb_tp_s()`_.
+
+A special note: as the definition and implementation of Go protect
+against dangling pointers in pure Go code, Go application code may not
+be designed and coded with the same level of defensiveness against
+dangling pointers that C applications are. In the case of
+:code:`TpS()`, owing the need to use :code:`unsafe pointer`
+parameters, please take additional care in designing and coding your
+application to ensure the validity of the pointers passed to
+:code:`TpS()`.
+
+--------------------------
+Go Simple API KeyT Methods
+--------------------------
+
+Go DataS()
+----------
+
+.. code-block:: go
+
+    DataS() int, int
+
+The :code:`DataS()` method returns the result of `ydb_data_s()`_ of
+the node referenced by the :code:`KeyT` structure, with a return code
+of :code:`C.YDB_ERR_OK` or an `error return code`_ prefixed with
+:code:`C.`. In the latter case, the return value is unspecified.
+
+Go DeleteS()
+------------
+
+.. code-block:: go
+
+    DeleteS(deltype int) int
```

Comment:
In order to allow use of the C.YDB_DEL_NODE/DEL_TREE constants to be used in this method, the type of 'deltype' should probably be C.int.